modules/git/repo_branch.go

```diff
@@ -48,6 +48,9 @@ type Branch struct {
 
 // GetHEADBranch returns corresponding branch of HEAD.
 func (repo *Repository) GetHEADBranch() (*Branch, error) {
+	if repo == nil {
```

Comment:
This unrelated change should not be here.

Reply:
@guillep2k schould I send a revert pull for https://github.com/go-gitea/gitea/pull/10401/commits/99867f3505d9e00dd40da7b471b6b0a1382fd370 ?

Reply:
When Tests failed a lot I add this so test dont crash only fail ...